src/Serializer.php

```diff
@@ -272,4 +277,24 @@ protected function getIncluded()
         $include = $this->request->get($this->expandParam);
         return is_string($include) ? preg_split('/\s*,\s*/', $include, -1, PREG_SPLIT_NO_EMPTY) : [];
     }
+
+    /**
+     * Checks Identification of resource object.
+     * http://jsonapi.org/format/#document-resource-object-identification
+     * @param ResourceInterface $resource
+     */
+    protected function checkIdentification(ResourceInterface $resource)
+    {
+        if (!is_string($resource->getId())) {
```

Comment:
Actually i think it should be responsibility of the `getId()` method. I think we just need to change the `ResourceTrait::getId()` a little bit. Instead of `return null` by default it should probably throw an error or return an empty string. https://github.com/tuyakhov/yii2-json-api/blob/master/src/ResourceTrait.php#L19
What do you think?